src/components/Chat/ChatSession.js

```diff
@@ -619,14 +702,20 @@ class ChatSession {
       newTranscript[lastDeliveredMessageIdx].lastDeliveredReceipt = true;
     }
 
+    // remove any view messages that are not the latest message
+    newTranscript = newTranscript.filter((item, idx) => {
+      return (idx === lastIncomingMessageIdx && lastOutgoingMessageIdx <= lastIncomingMessageIdx) || !(modelUtils.isViewMessage(item) &&
+        item.content.type === ContentType.MESSAGE_CONTENT_TYPE.INTERACTIVE_MESSAGE)
+    });
+
     this._updateTranscript(newTranscript);
   }
 
   _removePreviousInteractiveMessage(oldTranscript, newTranscript) {
     try {
       const newInteractiveMessage = newTranscript.find((message) => {
         const contentType = message.content.type;
-        return contentType === ContentType.MESSAGE_CONTENT_TYPE.INTERACTIVE_MESSAGE;
+        return contentType === ContentType.MESSAGE_CONTENT_TYPE.INTERACTIVE_MESSAGE && !modelUtils.isViewMessage(message);
```

Comment:
We don't do this here so that we can include the latest view message?

Reply:
This code is written for lex messages and is not applicable to views. This also causes unwanted effects such as removing the newest view when reloading a chat